ui/src/turing/components/form/standard_ensembler/LinkedRoutesTable.js

```diff
@@ -0,0 +1,166 @@
+  useEffect(() => {
+    if (isAllExperimentsLoaded) {
+      let newRouteToExperimentMappings = routes.reduce((m, r) => {m[r.id] = {running: {}, scheduled: {}}; return m}, {});
+      for (let experiment of allExperiments) {
+        for (let treatment of experiment.treatments) {
+          let configRouteName = getRouteName(treatment.configuration, treatmentConfigRouteNamePath);
+          if (typeof configRouteName === 'string' && configRouteName in newRouteToExperimentMappings) {
+            newRouteToExperimentMappings[configRouteName][getExperimentStatus(experiment).label.toLowerCase()][experiment.id] = experiment;
+          }
+        }
+      }
+      setRouteToExperimentMappings(newRouteToExperimentMappings);
+      setIsButtonPopoverOpen(routes.reduce((m, r) => {m[r.id] = {running: false, scheduled: false}; return m}, {}));
+    }
+  }, [treatmentConfigRouteNamePath, stringifiedRoutes, routes, isAllExperimentsLoaded, allExperiments]);
+
+  useEffect(() => {
+    if (isLoaded) {
+      if (!!experiments && !isAllExperimentsLoaded) {
+        setAllExperiments((curExperiments) => [...curExperiments, ...experiments]);
+      }
+      if (paging.pages > paging.page) {
+        setPageIndex(paging.page);
+      } else {
+        setIsAllExperimentsLoaded(true);
+      }
+    }
+  }, [isLoaded, experiments, paging, isAllExperimentsLoaded]);
+
+  const columns = [
+    {
+      field: "status",
+      width: "5px",
+      render: (_, item) => {
+        const isAssigned = routeToExperimentMappings[item.id] ?
+          Object.keys(routeToExperimentMappings[item.id].running).length +
+          Object.keys(routeToExperimentMappings[item.id].scheduled).length > 0 : false;
+        return (
+          <EuiIcon
+            type={isAssigned ? "check" : "cross"}
+            color={isAssigned ? "success" : "danger"}
+            size={"m"}
+            style={{ verticalAlign: "sub" }}
+          />
+        );
+      },
+    },
+    {
+      field: "route_name",
+      width: "20%",
+      name: "Route Name",
+      render: (_, item) => {
+        const isAssigned = routeToExperimentMappings[item.id] ?
+          Object.keys(routeToExperimentMappings[item.id].running).length +
+          Object.keys(routeToExperimentMappings[item.id].scheduled).length > 0 : false;
+        return (<EuiTextColor color={isAssigned ? "success" : "danger"}>{item.id}</EuiTextColor>);
+      },
+    },
+    {
+      field: "running_experiments",
+      width: "35%",
+      name: "Running Experiments",
+      render: (_, item) => (
+        <LinkedExperimentsContextMenu
+          item={item}
+          projectId={projectId}
+          routeToExperimentMappings={routeToExperimentMappings}
+          isButtonPopoverOpen={isButtonPopoverOpen}
+          setIsButtonPopoverOpen={setIsButtonPopoverOpen}
+          experimentStatus={"running"}
+        />
+      ),
+    },
+    {
+      field: "scheduled_experiments",
```

Comment:
How is this "scheduled_experiments" key computed and used by the `EuiInMemoryTable`?

Reply:
They aren't 😅 It's just a dummy and ultimately all the values get computed on the fly by the function passed to the `render` prop.